### PR TITLE
fix: Corrects errors when plan contains no pricing changes

### PR DESCRIPTION
**Pull request by mkubilus**
## Motivation and Context

This PR fixes a bug where if a plan would result in no pricing changes, but still have other resource updates a failure occurs:

```
│ Error: Error in function call
│ 
│   on .terraform/modules/pricing/modules/pricing/state_and_plan_content_from_json.tf line 59, in locals:
│   59:   instances_stripped_keys = local.input_type == "empty" ? [] : concat([
│   60:     for k, v in local.instances_map :
│   61:     setproduct([k], keys(v))
│   62:   ]...)
│     ├────────────────
│     │ local.instances_map is object with no attributes
│ 
│ Call to function "concat" failed: at least one argument is required.
```
With this update instead you will correctly receive pricing changes of 0:

```
Changes to Outputs:
  + pricing_input_resources         = {}
  + pricing_pricing_per_resources   = {}
  + pricing_pricing_product_filters = {}
  + pricing_resource_quantity       = {}
  + pricing_resources               = {}
  + pricing_total_price_per_hour    = 0
  + pricing_total_price_per_month   = 0
```

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: modules/pricing/state_and_plan_content_from_json.tf [hunk 1]
```diff
@@ -55,7 +55,7 @@ locals {
 
   instances_map = local.input_type == "plan" ? local.normalized_plan_instances : local.normalized_state_instances
 
-  instances_stripped_keys = local.input_type == "empty" ? [] : concat([
+  instances_stripped_keys = local.instances_map == {} ? [] : concat([
```
Comment (antonbabenko) [inline modules/pricing/state_and_plan_content_from_json.tf]:
```suggestion
  instances_stripped_keys = local.input_type == "empty" || length(local.instances_map) == 0 ? [] : concat([
```

I think this is a safer option. Do you agree?

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Thanks for the PR! A minor comment and we are good to go.

**Comment by mkubilus** [comment]
> Thanks for the PR! A minor comment and we are good to go.

I've tested this suggestion and verified that it resolves the original issue.  Thanks!

**Comment by antonbabenko** [comment]
This PR is included in version 1.3.1 :tada:

**Pull request merged**